### PR TITLE
main page design layout

### PR DESCRIPTION
**Pull request by e-garbuzov-wise**

### CODE REVIEW

File: Instagram-like-trainee/MainPage/View/CollectionViewCell/PostCell/PostHeaderCell.swift [hunk 1]
```diff
@@ -9,11 +9,58 @@ import UIKit
 
 class PostHeaderCell: UICollectionViewCell {
```
Comment (EugeneKirilovIOSDev) [inline Instagram-like-trainee/MainPage/View/CollectionViewCell/PostCell/PostHeaderCell.swift:10]: Same

File: Instagram-like-trainee/MainPage/View/MainPageViewController.swift [hunk 2]
```diff
@@ -44,10 +52,10 @@ class MainPageViewController: UIViewController {
...
     private func createHeaderLayout() -> NSCollectionLayoutSection {
-        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(100))
+        let itemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(75))
         let headerItem = NSCollectionLayoutItem(layoutSize: itemSize)
         
-        let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(100))
+        let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(75))
         let headerGroup = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, repeatingSubitem: headerItem, count: 1)
         
         let headerSection = NSCollectionLayoutSection(group: headerGroup)
```
Comment (EugeneKirilovIOSDev) [inline Instagram-like-trainee/MainPage/View/MainPageViewController.swift:61]: Please use methods of UICollectionView data source and delegate, just rewrite it, not good solution for now

File: Instagram-like-trainee/MainPage/View/MainPageViewController.swift [hunk 3]
```diff
@@ -56,24 +64,25 @@ class MainPageViewController: UIViewController {
...
-        let postActionItem = NSCollectionLayoutItem(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(50)))
+        let postActionItem = NSCollectionLayoutItem(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(35)))
         let postCaptionItem = NSCollectionLayoutItem(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(35)))
-        let postCommentSectionItem = NSCollectionLayoutItem(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(50)))
-        
+        let postCommentSectionItem = NSCollectionLayoutItem(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(20)))
         
         let feedGroup = NSCollectionLayoutGroup.vertical(layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(500)), subitems: [postHeaderItem, postImageItem, postActionItem, postCaptionItem, postCommentSectionItem])
```
Comment (EugeneKirilovIOSDev) [inline Instagram-like-trainee/MainPage/View/MainPageViewController.swift:88]: Here also, find some methods of delegate and data source

File: Instagram-like-trainee/MainPage/View/MainPageViewController.swift [hunk 4]
```diff
@@ -88,6 +97,7 @@ extension MainPageViewController: UICollectionViewDelegate, UICollectionViewData
     func numberOfSections(in collectionView: UICollectionView) -> Int {
```
Comment (EugeneKirilovIOSDev) [inline Instagram-like-trainee/MainPage/View/MainPageViewController.swift:97]: Don't forget to use MARK before extensions and for some code blocks separation

File: Instagram-like-trainee/MainPage/View/MainPageViewController.swift [hunk 5]
```diff
@@ -97,32 +107,45 @@ extension MainPageViewController: UICollectionViewDelegate, UICollectionViewData
...
             
             if indexPath.row == 0 {
-                cell.backgroundColor = .brown
+                let cell = collectionView.dequeueReusableCell(withReuseIdentifier: PostHeaderCell.identifier, for: indexPath)
+                return cell
             } else if indexPath.row == 1 {
-                cell.backgroundColor = .red
+                let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "feed", for: indexPath)
+                let image = UIImageView(image: UIImage(named: "elon-musk"))
+                image.contentMode = .scaleAspectFill
```
Comment (EugeneKirilovIOSDev) [inline Instagram-like-trainee/MainPage/View/MainPageViewController.swift:135]: Too much if operators, try to find better solution, use generative AI for code refactoring

**Review by EugeneKirilovIOSDev — CHANGES_REQUESTED** [review]

File: Instagram-like-trainee/Assets.xcassets/insta-logo.imageset/Instagram_logo.svg-2.png
[the remark this answers is not in the document]
Comment (e-garbuzov-wise) [reply Instagram-like-trainee/Assets.xcassets/insta-logo.imageset/Instagram_logo.svg-2.png:1]: fix

File: Instagram-like-trainee/Assets.xcassets/elon-musk.imageset/gettyimages-1229892983-square.jpg
[the remark this answers is not in the document]
Comment (e-garbuzov-wise) [reply Instagram-like-trainee/Assets.xcassets/elon-musk.imageset/gettyimages-1229892983-square.jpg:1]: fix

File: Instagram-like-trainee/MainPage/View/CollectionViewCell/HomeFeedHeaderCell.swift [hunk 6]
```diff
@@ -9,11 +9,53 @@ import UIKit
 
 class HomeFeedHeaderCell: UICollectionViewCell {
     static let identifier = "HomeFeedHeaderCell"
+    
+    let logoIamgeView = UIImageView()
```
[the remark this answers is not in the document]
Comment (e-garbuzov-wise) [reply Instagram-like-trainee/MainPage/View/CollectionViewCell/HomeFeedHeaderCell.swift:13]: fix

File: Instagram-like-trainee/MainPage/View/CollectionViewCell/HomeFeedHeaderCell.swift [hunk 7]
```diff
@@ -9,11 +9,53 @@ import UIKit
 
 class HomeFeedHeaderCell: UICollectionViewCell {
```
[the remark this answers is not in the document]
Comment (e-garbuzov-wise) [reply Instagram-like-trainee/MainPage/View/CollectionViewCell/HomeFeedHeaderCell.swift:10]: fix

File: Instagram-like-trainee/MainPage/View/CollectionViewCell/PostCell/PostActionCell.swift [hunk 8]
```diff
@@ -9,11 +9,68 @@ import UIKit
 
 class PostActionCell: UICollectionViewCell {
```
[the remark this answers is not in the document]
Comment (e-garbuzov-wise) [reply Instagram-like-trainee/MainPage/View/CollectionViewCell/PostCell/PostActionCell.swift:10]: fix

**Pull request merged**